tests/node/test_wakunode_store.nim

```diff
@@ -0,0 +1,662 @@
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 2
+        )
+
+        # When making the next history query
+        let queryResponse3 = await client.query(historyQuery3, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse3.get().messages == archiveMessages[4..<6]
+        
+        # Given the next query (4/5)
+        let historyQuery4 = HistoryQuery(
+          cursor: queryResponse3.get().cursor,
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 2
+        )
+
+        # When making the next history query
+        let queryResponse4 = await client.query(historyQuery4, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse4.get().messages == archiveMessages[6..<8]
+
+        # Given the next query (5/5)
+        let historyQuery5 = HistoryQuery(
+          cursor: queryResponse4.get().cursor,
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 2
+        )
+
+        # When making the next history query
+        let queryResponse5 = await client.query(historyQuery5, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse5.get().messages == archiveMessages[8..<10]
+      
+      asyncTest "Pagination with Large Page Size":
+        # Given the first query (1/2)
+        historyQuery.pageSize = 8
+
+        # When making a history query
+        let queryResponse1 = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse1.get().messages == archiveMessages[0..<8]
+        
+        # Given the next query (2/2)
+        let historyQuery2 = HistoryQuery(
+          cursor: queryResponse1.get().cursor,
+          pubsubTopic: some(pubsubTopic),
+          contentTopics: contentTopicSeq,
+          ascending: true,
+          pageSize: 8
+        )
+
+        # When making the next history query
+        let queryResponse2 = await client.query(historyQuery2, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse2.get().messages == archiveMessages[8..<10]
+        
+      asyncTest "Pagination with Excessive Page Size":
+        # Given the first query (1/1)
+        historyQuery.pageSize = 100
+
+        # When making a history query
+        let queryResponse1 = await client.query(historyQuery, serverRemotePeerInfo)
+
+        # Then the response contains the messages
+        check:
+          queryResponse1.get().messages == archiveMessages[0..<10]
+
+      asyncTest "Pagination with Zero Page Size":
+        # Given the first query (1/1)
+        historyQuery.pageSize = 0
```

Comment:
What does page size 0 mean? Is it something defined?

Reply:
Very interesting question and test :partying_face: !
@AlejandroCabeza - I think we shouldn't allow pageSize == 0

https://rfc.vac.dev/spec/13/

![image](https://github.com/waku-org/nwaku/assets/128452529/138b6631-7875-4fe8-abe0-8f053740e187)

Reply:
I was considering that to be equivalent to `None`, a.k.a. "give me everything".
Should we ask some RFC guy?

Reply:
Mmm. I definitely think we'd want to be explicit here in future versions of the RFC, but in protobuf if you leave an int field unset it would default to `0`. In other words, if someone leaves the `pageSize` unset it would be interpreted as page size `0` and nwaku would just use the default page size: `DefaultPageSize` - this seems reasonable to me.